test_utils/utils/setup_fixtures.py

```diff
@@ -14,6 +16,35 @@ def get_fixtures_data_from_file(filename):
 			return json.load(f)
 
 
+def before_test(company):
+	frappe.clear_cache()
+	today = frappe.utils.getdate()
+	setup_data = get_fixtures_data_from_file("setup.json")
+	companies = [d.get("company_name") for d in setup_data]
+
+	if company not in companies:
+		frappe.throw(f"Company: {company} does not exist in setup data.")
+
+	for setup in setup_data:
+		if company == setup.get("company_name"):
+			setup.update(
+				{
+					"fy_start_date": today.replace(month=1, day=1).isoformat(),
+					"fy_end_date": today.replace(month=12, day=31).isoformat(),
+				}
+			)
+			setup_complete(setup)
+			frappe.db.commit()
+	enable_all_roles_and_domains()
```

Comment:
I think we want to remove both `enable_all_roles_and_domains()` and `set_defaults_for_tests()` - they don't really apply to how we do testing and if they're required we can handle it in the project(s).